### PR TITLE
chore: migrate to secret token

### LINKED ISSUE #327: cannot validate webhooks (401 error) and therefore cannot run app
Hi, I'm trying to setup a chat app and it looks like the Zoom documentation has been updated since the README was last updated. Basically, adding an Event Subscription has "Validate" buttons now to validate the webhook. These do not work and give 401 errors. Unfortunately, this means I cannot test the app locally because it requires webhooks to be validated before I can hit the "Add" button. 

image

Which leads to...

image

As a heads up! The goal is to make a fun contribution to this project soon, using a Node.js version of https://github.com/ngrok/ngrok-go (coming soon to an npm package near you) to embed ngrok tunnel creation in chatbot.js :) 

<details><summary>Here's my stdout</summary>
[17:31:25.321] INFO (14960): Server listening at http://0.0.0.0:3000
[17:33:51.369] INFO (14960): incoming request
    reqId: "req-1"
    req: {
      "method": "POST",
      "url": "/hook",
      "hostname": "devgro-zoom-shuffle.ngrok.io",
      "remoteAddress": "127.0.0.1",
      "remotePort": 55419
    }
[17:33:51.371] INFO (14960): Unauthorized
    reqId: "req-1"
    res: {
      "statusCode": 401
    }
    err: {
      "type": "UnauthorizedError",
      "message": "Unauthorized",
      "stack":
          UnauthorizedError: Unauthorized
              at verifyRequest (file:///Users/celina/ngrok-stuff/zoom-shuffle-bot/plugins/verifyRequest.js:10:11)
              at hookIterator (/Users/celina/ngrok-stuff/zoom-shuffle-bot/node_modules/fastify/lib/hooks.js:246:10)
              at next (/Users/celina/ngrok-stuff/zoom-shuffle-bot/node_modules/fastify/lib/hooks.js:174:16)
              at hookRunner (/Users/celina/ngrok-stuff/zoom-shuffle-bot/node_modules/fastify/lib/hooks.js:196:3)
              at Object.routeHandler [as handler] (/Users/celina/ngrok-stuff/zoom-shuffle-bot/node_modules/fastify/lib/route.js:469:7)
              at Router.callHandler (/Users/celina/ngrok-stuff/zoom-shuffle-bot/node_modules/find-my-way/index.js:413:14)
              at Router.lookup (/Users/celina/ngrok-stuff/zoom-shuffle-bot/node_modules/find-my-way/index.js:391:17)
              at Server.preRouting (/Users/celina/ngrok-stuff/zoom-shuffle-bot/node_modules/fastify/fastify.js:773:14)
              at Server.emit (node:events:513:28)
              at parserOnIncoming (node:_http_server:1061:12)
      "status": 401,
      "statusCode": 401,
      "expose": true
    }
[17:33:51.380] INFO (14960): request completed
    reqId: "req-1"
    res: {
      "statusCode": 401
    }
    responseTime: 10.50583267211914

</details>

### LINKED ISSUE #328: Webhooks are not validated
Related to #327 . It looks like this repo was set up before was doing webhooks validation. It would be good, for the sake of security to set it up. The samples I posted in the comments to that PR are in an internal repo, ask to get access please.

### PR DESCRIPTION
**Pull request by RomuloVitoi**
Zoom deprecated the verification token in favor of a secret token that is used to sign the requests.
It also now requires that a `endpoint.url_validation` event is responded by the webhook during the setup to validate that you're doing the signature check.

The `ZOOM_SECRET_TOKEN` secret must be added to this repo settings.

Fixes #327
Fixes #328

### CODE REVIEW

**Review by simoneb — COMMENTED** [review]
I'm going through this PR (there are lots of changes!). Meanwhile, please note:

- the gcloud secrets need to be created manually (as described in the docs) because we don't have infra provisioning code in this repo
- as a consequence of that, the docs also need to be updated to reflect this, as well as (I suppose) some of the screenshots (we can do this together)

**Comment by RomuloVitoi** [comment]
The CI updates the secrets if they're different but since `gcloud secrets versions access` exits with code 1 if the secret does not exist I'm not entirely sure that it will continue executing.

**Comment by RomuloVitoi** [comment]
The secret was created in GCP and set in the repo settings.

File: plugins/sendBotMessage.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ export default async function sendBotMessage(
     },
     body: JSON.stringify({
       robot_jid: options.botJid,
+      user_jid: userJid,
```
Comment (simoneb) [inline plugins/sendBotMessage.js:44]: what's this change?

File: routes/bot.js [hunk 2]
```diff
@@ -6,16 +6,17 @@ import sortRandomly from '../helpers/sortRandomly.js'
 export default async function (fastify) {
   fastify.post(
     '/bot',
-    { onRequest: [fastify.zoom.verifyRequest] },
+    { preHandler: [fastify.zoom.verifyRequest] },
```
Comment (simoneb) [inline routes/bot.js:9]: why the change in the hook?

**Review by simoneb — APPROVED** [review]
a couple of comments but overall LGTM, very nice job 👌

File: routes/bot.js (hunk 2 above)
Comment (RomuloVitoi) [reply routes/bot.js:9]: the request body is not available in the `onRequest` hook: https://www.fastify.io/docs/latest/Reference/Hooks/#onrequest

File: plugins/sendBotMessage.js (hunk 1 above)
Comment (RomuloVitoi) [reply plugins/sendBotMessage.js:44]: The requests to send messages were failing with a "Invalid request body format" message. I've found a few references in their dev forum pointing to this field that should be added, probably a new field that has become required I believe: https://marketplace.zoom.us/docs/guides/team-chat-apps/send-edit-and-delete-messages/#send-messages

**Comment by simoneb** [comment]
Thanks for the clarifications, merge away 👌
